src/NServiceBus.RabbitMQ/TaskEx.cs

```diff
@@ -1,10 +1,20 @@
 namespace NServiceBus.Transport.RabbitMQ
 {
+    using System;
+    using System.Threading;
     using System.Threading.Tasks;
 
     static class TaskEx
     {
         //TODO: remove when we update to 4.6 and can use Task.CompletedTask
         public static readonly Task CompletedTask = Task.FromResult(0);
+
+        public static Task StartNew(object state, Action<object> action) => StartNew(state, action, TaskScheduler.Default);
+
+        public static Task StartNew(object state, Action<object> action, TaskScheduler scheduler) => Task.Factory.StartNew(action, state, CancellationToken.None, TaskCreationOptions.DenyChildAttach, scheduler);
+
+        public static Task StartNew(object state, Func<object, Task> function) => StartNew(state, function, TaskScheduler.Default);
+
+        public static Task StartNew(object state, Func<object, Task> function, TaskScheduler scheduler) => Task.Factory.StartNew(function, state, CancellationToken.None, TaskCreationOptions.DenyChildAttach, scheduler).Unwrap();
```

Comment:
I like the simplification this affords in `MessagePump` but I think only the 2nd overload should exist as it's the only one that's required. The 1st, 3rd and 4th overloads aren't used so I'd call YAGNI on them and remove them. If we find the further overloads are needed later it's trivial to add them.

Also, given the method is only used in `MessagePump`, I'd probably say YAGNI on it being internal and make it a private in that class.

BTW - I think this would also work:

```C#
static Task StartNew<T>(T state, Action<T> action, TaskScheduler scheduler) =>
    Task.Factory.StartNew(
        o => action((T)o), state, CancellationToken.None, TaskCreationOptions.DenyChildAttach, scheduler);
```

It saves having to do the cast in the delegate that's passed into the method.